gpt_engineer/applications/cli/file_selector.py

```diff
@@ -336,112 +275,143 @@ def ask_for_files(project_path: Union[str, Path]) -> FilesDict:
     else:
-        use_last_string = ""
         if FILE_LIST_NAME in metadata_db:
-            use_last_string = (
-                "3. Use previous file list (available at "
-                + f"{os.path.join(metadata_db.path, FILE_LIST_NAME)})\n"
+            print(
+                f"File list detected at {metadata_db.path / FILE_LIST_NAME}. "
+                "Edit or delete it if you want to select new files."
             )
-            selection_number = 3
+            selected_files = editor_file_selector(project_path, False)
         else:
-            selection_number = 1
-        selection_str = "\n".join(
-            [
-                "How do you want to select the files?",
-                "",
-                "1. Use File explorer.",
-                "2. Use Command-Line.",
-                use_last_string if len(use_last_string) > 1 else "",
-                f"Select option and press Enter (default={selection_number}): ",
-            ]
-        )
-
-        file_path_list = []
-        selected_number_str = input(selection_str)
-        if selected_number_str:
-            try:
-                selection_number = int(selected_number_str)
-            except ValueError:
-                print("Invalid number. Select a number from the list above.\n")
-                sys.exit(1)
-
-        if selection_number == 1:
-            # Open GUI selection
-            file_path_list = gui_file_selector(project_path)
-        elif selection_number == 2:
-            # Open terminal selection
-            file_path_list = terminal_file_selector(project_path)
-        if (
-            selection_number <= 0
-            or selection_number > 3
-            or (selection_number == 3 and not use_last_string)
-        ):
-            print("Invalid number. Select a number from the list above.\n")
-            sys.exit(1)
-
-        # ToDO: Replace this hack that makes all file_path relative to the right thing
-        file_path_list = [
-            os.path.relpath(file_path.strip(), project_path)
-            for file_path in file_path_list
-        ]
-
-        if not selection_number == 3:
-            metadata_db[FILE_LIST_NAME] = "\n".join(
-                str(file_path) for file_path in file_path_list
-            )
-    content_dict = dict()
-    with open(metadata_db.path / FILE_LIST_NAME, "r") as file_list:
-        for file in file_list:
-            with open(os.path.join(project_path, file.strip()), "r") as content:
-                content_dict[file.strip()] = content.read()
+            selected_files = editor_file_selector(project_path, True)
+    content_dict = {}
+    for file_path in selected_files:
+        # selected files contains paths that are relative to the project path
+        file_path = Path(file_path)
+        try:
+            # to open the file we need the path from the cwd
+            with open(Path(project_path) / file_path, "r") as content:
```

Comment:
Since file path is now relative to the project directory, we have to add the project path to read the right file